From 11ac41dcd6831e983354bc7f8d0eeaf848adcc6d Mon Sep 17 00:00:00 2001
From: DEEXORT <deexort@mail.ru>
Date: Thu, 2 Mar 2017 11:27:46 +0300
Subject: [PATCH] Fix signs in pose calculation

---
 src/turtle.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/turtle.cpp b/src/turtle.cpp
index 4eef18f..8798141 100644
--- a/src/turtle.cpp
+++ b/src/turtle.cpp
@@ -152,9 +152,9 @@ bool Turtle::update(double dt, QPainter& path_painter, const QImage& path_image,
   QPointF old_pos = pos_;
 
   orient_ = std::fmod(orient_ + ang_vel_ * dt, 2*PI);
-  pos_.rx() += std::cos(orient_) * lin_vel_x * dt +
+  pos_.rx() += std::cos(orient_) * lin_vel_x * dt -
                std::sin(orient_) * lin_vel_y * dt;
-  pos_.ry() += -std::sin(orient_) * lin_vel_x * dt +
+  pos_.ry() += -std::sin(orient_) * lin_vel_x * dt -
                 std::cos(orient_) * lin_vel_y * dt;
 
   // Clamp to screen size
-- 
1.9.1

